aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -56,6 +56,24 @@ Tensor _to_copy(
   bool pin_out = (non_blocking && self.is_cuda() && options.device().is_cpu() &&
                   (options.layout() == c10::kStrided));
```

Comment:
`pin_out` is unused if `self.is_sparse_csr()` when False. Hence, move its definition after the `if (self.is_sparse()) {...}` block.